hashing/te-tag-query-java/com/facebook/threatexchange/DescriptorPostParameters.java

```diff
@@ -330,60 +330,58 @@ public String getReactionsToRemove() {
     return this._reactionsToRemove;
   }
 
-  public boolean validateForSubmitWithReport(PrintStream o) {
+  // Returns null if no errors, non-null for error.
+  public String validateForSubmitWithReport() {
     if (this._descriptorID != null) {
-      o.println("Descriptor ID must not be specified for submit.\n");
-      return false;
+      return "Descriptor ID must not be specified for submit.";
     }
     if (this._indicatorText == null) {
-      o.println("Indicator text is missing.\n");
-      return false;
+      return "Indicator text is missing.";
```

Comment:
nit: "must be specified for x." vs "is missing." Could be good to have consistent language in the error messages and have a slight preference for the former. 

Additionally, it could be nice to surface all the missing fields at once. "[x], [y], & [z] must be specified for [action]." (or "are missing.")

Reply:
1. nice

2. yes ... let's do that as a separate commit, and for all three lang impls.